packages/tupaia-web/src/api/queries/useMapOverlayReport.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+import { useQuery } from 'react-query';
+import { momentToDateString } from '@tupaia/utils';
+import {
+  autoAssignColors,
+  createValueMapping,
+  getSpectrumScaleValues,
+  SPECTRUM_MEASURE_TYPES,
+} from '@tupaia/ui-map-components';
+import { get } from '../api';
+import { EntityCode, ProjectCode, SingleMapOverlayItem } from '../../types';
+
+// make the response from the new endpoint look like the response from the legacy endpoint
+const normaliseResponse = (measureDataResponse: any, overlay: SingleMapOverlayItem) => {
+  const { measureCode, measureLevel, displayType, dataElementCode, ...restOfOverlay } = overlay;
+
+  const measureOptions = [
+    {
+      measureLevel,
+      type: displayType,
+      key: dataElementCode || 'value',
+      ...restOfOverlay,
+    },
+  ];
+
+  return {
+    measureCode,
+    measureLevel,
+    measureOptions,
+    serieses: measureOptions,
+    measureData: measureDataResponse,
+  };
+};
+
```

Comment:
I don't see anything wrong with this, but I guess it opens up the question, should we be doing this on the frontend, or should the endpoint be returning data the correct way? Or is this because dashboard reports and map overlay reports are both using the `report` endpoint?

Reply:
Yes good question. I just had a chat to Ethan about this and he said that this could be done on the backend but it would be messy because we use the report route is used fro both the map overlays and the dashboard items.